packages/kbn-es-archiver/src/lib/indices/create_index_stream.ts

```diff
@@ -129,6 +135,13 @@ export function createCreateIndexStream({
       return;
     }
 
+    if (isSavedObjectIndex(index) && !isArchiveInExceptionList) {
+      throw new Error(
+        `esArchiver doesn't support modifying the existing Saved Objects index: '${index}',
+      please update its definition in mappings.json`
```

Comment:
@gsoldevila is it a good request, or better "please remove its definition" ?

Reply:
might be worth mentioning removal from data.json in addition to mappings.json, and what actions to take instead.

Reply:
In this specific scenario folks are creating a new archive. I'd rather go with something like:

```
ERROR: 'esArchiver' no longer supports defining saved object indices.
The recommendation is to use 'kbnArchiver' to import saved objects in your tests.
If you absolutely need to load some non-importable SOs, please stick to the official saved object indices created by Kibana at startup.
You can achieve that by simply removing your saved object index definitions from 'mappings.json' (likely removing the file altogether).
```

Then, for the whitelisted ones, it's about awareness for existing tests. A possible warning message could be:
```
WARNING: This test is using an 'esArchiver' archive that contains saved object index definitions (in the 'mappings.json').
This has proven to be a source of conflicts and flakiness, so the goal is to remove support for this feature ASAP.
We kindly ask you to update your test archives and remove SO index definitions, so that tests use the official saved object indices created by Kibana at startup.
You can achieve that by simply removing your saved object index definitions from 'mappings.json' (likely removing the file altogether).
We also recommend migrating existing tests to 'kbnArchiver' whenever possible.
```

I would also add a link to ?GH issue? where we can put more information to guide devs.

WDYT?

Reply:
UPDATE: Sorry I just realised you already had a warning message. Asking to update the archives and removing them from the whitelist seems like a good idea 🔝 . We can probably merge the messages.